include/boost/wave/util/cpp_macromap.hpp

```diff
@@ -1004,13 +1023,56 @@ bool adjacent_stringize = false;
                     impl::replace_ellipsis(expanded_args, i, expanded, pos);
                 }
                 else
+
+#if BOOST_WAVE_SUPPORT_VA_OPT != 0
+                if (is_va_opt) {
+                    position_type const &pos = (*cit).get_position();
+
+                    BOOST_ASSERT(boost::wave::need_va_opt(ctx.get_language()));
+
+                    // ensure all variadic arguments to be expanded
+                    for (typename vector<ContainerT>::size_type arg = i;
+                         arg < expanded_args.size(); ++arg)
+                    {
+                        expand_argument(arg, arguments, expanded_args,
+                            expand_operator_defined, has_expanded_args);
+                    }
+
+                    // locate the __VA_OPT__ arguments
+                    typename macro_definition_type::const_definition_iterator_t cstart = cit;
+                    if (!impl::find_va_opt_params(cit, cend)) {
+                        BOOST_WAVE_THROW_CTX(ctx, preprocess_exception,
+                            improperly_terminated_macro, "missing ')' in __VA_OPT__",
+                            pos);
+                    }
+                    // cstart still points to __VA_OPT__; cit now points to the last rparen
+
+                    if ((i == arguments.size()) ||             // no variadic argument
+                        impl::is_blank_only(arguments[i])) {   // no visible tokens
+                        // no args; insert placemarker
+                        *std::back_inserter(expanded) =
+                            typename ContainerT::value_type(T_PLACEMARKER, "\xA7", pos);
```

Comment:
Also, this inserts a place marker even if the token sequence already contains one. Is that intended?

Reply:
1. Yes :) I will make the change.
2. I'm not sure how to avoid inserting two placemarkers. Is there an appropriate Wave idiom, or a different technique I should use?

Reply:
> 2. I'm not sure how to avoid inserting two placemarkers. Is there an appropriate Wave idiom, or a different technique I should use?

I think using `is_whitespace_only` instead of `is_blank_only` would be sufficient to resolve this.

Reply:
Well, unfortunately making that change causes two of my test cases to fail: [this one](https://github.com/jefftrull/wave/blob/9121d11e20adb392160f2d4ec8f19a64a901677b/test/testwave/testfiles/t_8_001.cpp#L29) and [this one](https://github.com/jefftrull/wave/blob/9121d11e20adb392160f2d4ec8f19a64a901677b/test/testwave/testfiles/t_8_001.cpp#L43). And that's why the code was change to test for placemarkers... Why that was necessary I'm not sure and I would not be surprised if there was a better solution (or a bug I introduced elsewhere).

Reply:
Ok, I'll investigate this further.

Reply:
How about writing it as:
```c++
                    if ((i == arguments.size()) ||             // no variadic argument
                        impl::is_whitespace_only(arguments[i])) {   // no visible tokens
                        // no args; insert placemarker if one is not already present
                        expanded.push_back(
                            typename ContainerT::value_type(T_PLACEMARKER, "\xA7", pos));
                    }
                    else if (!impl::is_blank_only(arguments[i])) {
                        // …
                    }
```